### PR TITLE
install support download with apt-helper

### PR DESCRIPTION
**Pull request by h0tw1r3**
recent ubuntu official docker images are unable to download with any of the other methods

### CODE REVIEW

File: tasks/install_shell.sh [hunk 1]
```diff
@@ -432,6 +437,19 @@ do_fetch() {
   return 0
 }
 
+do_apt_helper() {
+  info "Trying apt-helper..."
+  run_cmd "/usr/lib/apt/apt-helper download-file \"$1\" \"$2\"" 2>$tmp_stderr
```
Comment (bastelfreak) [inline tasks/install_shell.sh:442]: Do we need double quotes? 
```suggestion
  run_cmd "/usr/lib/apt/apt-helper download-file '$1' '$2'" 2>$tmp_stderr
```

Comment (h0tw1r3) [reply tasks/install_shell.sh]: updated

**Review by bastelfreak — APPROVED** [review]

File: tasks/install_shell.sh [hunk 2]
```diff
@@ -525,6 +543,10 @@ do_download() {
     do_python3_urllib $1 $2 && return 0
   fi
 
+  if exists_apt_helper; then
+    do_apt_helper $1 $2 && return 0
+  fi
+
   critical "Cannot download package as none of wget/curl/fetch/perl-LWP-Simple/perl-File-Fetch/python3 is found"
```
Comment (kenyon) [inline tasks/install_shell.sh:550]:
```suggestion
  critical "Cannot download package as none of wget/curl/fetch/perl-LWP-Simple/perl-File-Fetch/python3/apt-helper is found"
```

**Review by kenyon — CHANGES_REQUESTED** [review]

File: tasks/install_shell.sh (hunk 2 above)
Comment (h0tw1r3) [reply tasks/install_shell.sh]: updated

**Review by kenyon — APPROVED** [review]

File: tasks/install_shell.sh [hunk 3]
```diff
@@ -432,6 +437,19 @@ do_fetch() {
...
+  info "Trying apt-helper..."
+  run_cmd "/usr/lib/apt/apt-helper download-file '$1' '$2'" 2>$tmp_stderr
+  rc=$?
+
+  if test $rc -eq 0 && test -s "$2" ; then
+    return 0
+  fi
+
+  capture_tmp_stderr "apthelper"
+  return 1
```
Comment (joshcooper) [inline tasks/install_shell.sh:450]: All of the previous `do_wget/curl/etc` test for failures like

```
if test $rc -ne 0 || test ! -s "$2"; then
  return 1
fi
```
and then `return 0` at the end. Could you do the same here?

Comment (h0tw1r3) [reply tasks/install_shell.sh]: done, also added 404 test like other methods

**Comment by joshcooper** [comment]
Closed and reopened to kick tests

**Pull request merged**